app/controllers/tags_controller.rb

```diff
@@ -213,4 +214,21 @@ def tag_params
   def exec_sql(sql_array)
     ApplicationRecord.connection.execute(ActiveRecord::Base.sanitize_sql_array(sql_array))
   end
+
+  def verify_tag_editor
+    if !user_signed_in? ||
+       !(current_user&.privilege?(:edit_tags) || current_user&.is_moderator || current_user&.is_admin)
```

Comment:
suggestion: use an `unless` block instead. This condition is hard to read naturally; switching the conditional to `unless` makes it a bit easier:

```suggestion
    unless user_signed_in? && (current_user.privilege?(:edit_tags) || current_user.is_moderator || current_user.is_admin)
```

Reply:
Applied and tested.